src/metadata-builder/RelationJoinColumnBuilder.ts

```diff
@@ -133,11 +133,13 @@ export class RelationJoinColumnBuilder {
             });
             const joinColumnName = joinColumnMetadataArg ? joinColumnMetadataArg.name : this.connection.namingStrategy.joinColumnName(relation.propertyName, referencedColumn.propertyName);
 
-            let relationalColumn = relation.entityMetadata.ownColumns.find(column => column.databaseName === joinColumnName);
+            const relationalColumns = relation.embeddedMetadata ? relation.embeddedMetadata.columns : relation.entityMetadata.ownColumns;
```

Comment:
what if there is an embedded, but we point to a relation that not in the embedded?

Reply:
I'm not sure what you mean, I think `relation.embeddedMetadata` is only set if `relation` is a column within an embedded entity. It was somewhat confusing initially but `relation` refers only to the foreign key column, not the target relation entity. 

This particular section is what searches for an existing relation id column, and if the relation is located inside an embedded then it only makes sense to search for the relation id column also in the embedded rather than in the base of the entity.

Reply:
okay, thank you for explanation

Reply:
I've had a second look at this, and in theory we could allow relation IDs to not be in the embedded entity together with the relation, but I'm not sure whether that would cause additional confusion.

```typescript
@Entity()
class Other {
    @PrimaryColumn() id: number;
}
@Entity()
class User {
    @Column(() => Embed) embed: Embed;

    @Column() embedOther1Id: number;
    @ManyToOne(() => Other) embedOther2: Other;
}
class Embed {
    @ManyToOne(() => Other) other1: Other;
    @Column() other2Id: number;
}
```

Both relations here could match up correctly if we want, and if they are not matched up they will lead to a duplicate column error anyway. MongoDB is an exception since it maintains the embedded objects in DB, so both relation/ID have to be in the same entity.